tests/virt/cluster/general/test_mass_machine_type_transition.py

```diff
@@ -0,0 +1,218 @@
+):
+    assert vm_with_schedulable_machine_type.vmi.status == VirtualMachineInstance.Status.RUNNING, (
+        f"VM {vm_with_schedulable_machine_type.name} with schedulable machine type is not running"
+    )
+    assert (
+        vm_with_unschedulable_machine_type.instance.status.get("printableStatus")
+        == VirtualMachine.Status.ERROR_UNSCHEDULABLE
+    ), f"VM {vm_with_unschedulable_machine_type.name} with unschedulable machine type should not be running"
+
+
+@pytest.mark.usefixtures(
+    "kubevirt_api_lifecycle_namespace",
+    "kubevirt_api_lifecycle_service_account",
+    "kubevirt_api_lifecycle_cluster_role_binding",
+)
+class TestMachineTypeTransition:
+    @pytest.mark.parametrize(
+        "kubevirt_api_lifecycle_automation_job",
+        [
+            pytest.param(
+                {
+                    "machine_type_glob": "pc-q35-rhel8.*.*",
+                    "restart_required": "true",
+                },
+                marks=pytest.mark.polarion("CNV-11949"),
+            ),
+        ],
+        indirect=True,
+    )
+    def test_machine_type_transition_with_restart_true(
+        self,
+        vm_with_schedulable_machine_type,
+        machine_type_from_kubevirt_config,
+        kubevirt_api_lifecycle_automation_job,
+    ):
+        wait_for_running_vm(vm=vm_with_schedulable_machine_type)
+        assert (
+            vm_with_schedulable_machine_type.vmi.instance.spec.domain.machine.type == machine_type_from_kubevirt_config
+        ), f"VM {vm_with_schedulable_machine_type.name} should have machine type {machine_type_from_kubevirt_config}"
+
+    @pytest.mark.parametrize(
+        "kubevirt_api_lifecycle_automation_job",
+        [
+            pytest.param(
+                {
+                    "machine_type_glob": "pc-q35-rhel8.*.*",
+                    "restart_required": "false",
+                },
+                marks=pytest.mark.polarion("CNV-11950"),
+            ),
+        ],
+        indirect=True,
+    )
+    def test_machine_type_transition_without_restart(
+        self,
+        vm_with_schedulable_machine_type,
+        machine_type_from_kubevirt_config,
+        kubevirt_api_lifecycle_automation_job,
+    ):
+        assert vm_with_schedulable_machine_type.vmi.status == VirtualMachineInstance.Status.RUNNING
+        assert (
+            vm_with_schedulable_machine_type.vmi.instance.spec.domain.machine.type == MachineTypesNames.pc_q35_rhel8_1
+        ), f"VM {vm_with_schedulable_machine_type.name} should have same machine type as before transition"
+        restart_vm_wait_for_running_vm(vm=vm_with_schedulable_machine_type, wait_for_interfaces=True)
+        assert (
+            vm_with_schedulable_machine_type.vmi.instance.spec.domain.machine.type == machine_type_from_kubevirt_config
+        ), f"VM {vm_with_schedulable_machine_type.name} should have machine type {machine_type_from_kubevirt_config}"
```

Comment:
how about smth like:
```suggestion
        ), f"Machine Type of the VM {vm_with_schedulable_machine_type.name} changed after restart, expected:  {machine_type_from_kubevirt_config}, current: {current_machine_type}"
```

Reply:
we do expect machine to update after restart,
for adding current machine type, how about I update it to
```
    assert machine_type_after_restart == machine_type_from_kubevirt_config, (
            f"VM {vm_with_schedulable_machine_type.name} should have machine type {machine_type_from_kubevirt_config},"
            f"current machine type: {machine_type_after_restart}"
```